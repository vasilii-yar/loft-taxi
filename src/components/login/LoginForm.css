.login-button {
    margin-top: 20px;
    margin-left: 80%;
}

.login-header {
    padding-top: 20px;
    padding-bottom: 20px;
}

.login-form-layout {
    width: 440px;
}

.login-content-container {
    padding-left: 40px;
    padding-right: 40px;
}