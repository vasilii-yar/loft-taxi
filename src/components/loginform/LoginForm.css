.login-button {
    margin-top: 20px;
    margin-left: 80%;
}

.login-header {
    padding-top: 20px;
    padding-bottom: 20px;
}