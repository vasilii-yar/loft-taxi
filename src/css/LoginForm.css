.login-button {
    margin-top: 50%;
}