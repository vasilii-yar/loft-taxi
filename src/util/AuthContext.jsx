import React, {useEffect} from 'react';

export const Context = React.createContext();

/*export default class AuthContext extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isLoggedIn: false
        };
    }

    logIn = (email, password) => {
        this.setState({isLoggedIn: true});
    }

    logOut = () => {
        this.setState({isLoggedIn: false});
    }

    isLoggedIn = () => {
        return this.state.isLoggedIn;
    }

    render() {
        return (
            <Context.Provider value={{
                logIn: this.logIn,
                logOut: this.logOut,
                isLoggedIn: this.isLoggedIn
            }}>
                {this.props.children}
            </Context.Provider>
        );
    }
}*/

export const AuthProvider = ({children}) => {
    const [isLoggedIn, setIsLoggedIn] = React.useState(false);

    const logIn = (email, password) => {
        setIsLoggedIn(true);
    }

    const logOut = () => {
        setIsLoggedIn(false);
    }

    return (
        <Context.Provider value={{logIn, logOut, isLoggedIn}}>
            {children}
        </Context.Provider>
    );
}

export const withAuth = (WrappedComponent) => {
    return class extends React.Component {
        render() {
            return (
                <Context.Consumer>
                    {
                        (value) => {
                            return <WrappedComponent {...value} {...this.props}/>
                        }
                    }
                </Context.Consumer>
            );
        }
    }
}
