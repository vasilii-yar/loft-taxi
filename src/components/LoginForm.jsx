import React from 'react';
import Paper from '@material-ui/core/Paper';
import TextField from '@material-ui/core/TextField';
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import Container from "@material-ui/core/Container";
import Link from "@material-ui/core/Link";
import "../css/LoginForm.css"

export default class LoginForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            login: "",
            password: ""
        }
    }

    handleSubmit = () => {

    }

    goToRegistration = () => {

    }

    handleChange = (event) => {
        const target = event.target;
        const name = target.name;
        const value = target.value;

        this.setState(
            {[name]: value}
        )
    }

    render() {
        return (
            <Paper>
                <Container>
                    <h1>Войти</h1>
                    <form onSubmit={this.handleSubmit}>
                        <Grid container direction={"column"}>
                            <Grid item xs={12}>
                                <Typography>
                                    Новый пользователь? <Link href={"#"}
                                                              onClick={this.goToRegistration}>Зарегистрируйтесь</Link>
                                </Typography>
                            </Grid>
                            <Grid item xs={12}>
                                <TextField label={"Имя пользователя"} type={"email"} onChange={this.handleChange}/>
                            </Grid>
                            <Grid item xs={12}>
                                <TextField label={"Пароль"} type={"password"} onChange={this.handleChange}/>
                            </Grid>
                            <Grid item xs={12}>
                                <Button className={"login-button"} variant="contained" color="primary" type="submit" className="button-block">
                                    Войти
                                </Button>
                            </Grid>
                        </Grid>
                    </form>
                </Container>
            </Paper>
        );
    }
}