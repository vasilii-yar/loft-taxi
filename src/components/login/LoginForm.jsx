import React from 'react';
import Paper from '@material-ui/core/Paper';
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import Container from "@material-ui/core/Container";
import "./LoginForm.css"
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import Input from "@material-ui/core/Input";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {tryAuth} from "../../redux/modules/auth/authActions";
import {Link, withRouter} from "react-router-dom";

class LoginForm extends React.Component {
    static propTypes = {
        tryAuth: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props);
        this.state = {
            login: "",
            password: ""
        }
    }

    handleSubmit = (event) => {
        event.preventDefault();
        this.props.tryAuth(this.state.login, this.state.password);
    }

    goToMap = () => {
        this.props.history.push("/map");
    }

    goToRegistration = () => {
        this.props.history.push("/registration");
    }

    handleChange = (event) => {
        const target = event.target;
        const name = target.name;
        const value = target.value;

        this.setState(
            {[name]: value}
        )
    }


    render() {
        return (
            this.props.isLoggedIn ?
                (
                    <>
                        {this.goToMap()}
                    </>
                )
                :
                (
                    <Paper>
                        <Container>
                            <Typography variant="h4" gutterBottom className="login-header">
                                Войти
                            </Typography>
                            <form onSubmit={this.handleSubmit}>
                                <Grid container direction="column" spacing={2}>
                                    <Grid item xs={12}>
                                        <Typography>
                                            Новый пользователь? <Link to="/registration">Зарегистрируйтесь</Link>
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormControl fullWidth>
                                            <InputLabel htmlFor="login">Пользователь</InputLabel>
                                            <Input
                                                id="login"
                                                name="login"
                                                type="email"
                                                value={this.state.login}
                                                onChange={this.handleChange}
                                                fullWidth
                                            />
                                        </FormControl>
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormControl fullWidth>
                                            <InputLabel htmlFor="password">Пароль</InputLabel>
                                            <Input
                                                id="password"
                                                name="password"
                                                type="password"
                                                value={this.state.password}
                                                onChange={this.handleChange}
                                                fullWidth
                                            />
                                        </FormControl>
                                    </Grid>
                                    <Grid item xs={12}>
                                        <div className="login-button">
                                            <Button variant="contained" color="primary" type="submit">
                                                Войти
                                            </Button>
                                        </div>
                                    </Grid>
                                </Grid>
                            </form>
                        </Container>
                    </Paper>
                )
        );
    }
}

export default connect(
    (state) => ({isLoggedIn: state.auth.isLoggedIn}),
    {tryAuth}
)(withRouter(LoginForm));