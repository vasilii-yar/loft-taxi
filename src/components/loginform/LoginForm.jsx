import React from 'react';
import Paper from '@material-ui/core/Paper';
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import Container from "@material-ui/core/Container";
import Link from "@material-ui/core/Link";
import "./LoginForm.css"
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import Input from "@material-ui/core/Input";
import PropTypes from "prop-types";
import {withAuth} from "../../util/AuthContext";

class LoginForm extends React.Component {
    static propTypes = {
        logIn: PropTypes.func.isRequired,
        navigateTo: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props);
        this.state = {
            login: "",
            password: ""
        }
    }

    handleSubmit = () => {
        this.props.logIn(this.state.login, this.state.password);
        this.props.navigateTo("map");
    }

    goToRegistration = () => {
        this.props.navigateTo("registration")
    }



    render() {
        return (
            <Paper>
                <Container>
                    <Typography variant="h4" gutterBottom className="login-header">
                        Войти
                    </Typography>
                    <form onSubmit={this.handleSubmit}>
                        <Grid container direction="column" spacing={2}>
                            <Grid item xs={12}>
                                <Typography>
                                    Новый пользователь? <Link href="#"
                                                              onClick={this.goToRegistration}>Зарегистрируйтесь</Link>
                                </Typography>
                            </Grid>
                            <Grid item xs={12}>
                                <FormControl fullWidth>
                                    <InputLabel htmlFor="login">Пользователь</InputLabel>
                                    <Input
                                        id="login"
                                        name="login"
                                        type="email"
                                        onChange={this.handleChange}
                                        fullWidth
                                    />
                                </FormControl>
                            </Grid>
                            <Grid item xs={12}>
                                <FormControl fullWidth>
                                    <InputLabel htmlFor="password">Пароль</InputLabel>
                                    <Input
                                        id="password"
                                        name="password"
                                        type="password"
                                        onChange={this.handleChange}
                                        fullWidth
                                    />
                                </FormControl>
                            </Grid>
                            <Grid item xs={12}>
                                <div className="login-button">
                                    <Button variant="contained" color="primary" type="submit">
                                        Войти
                                    </Button>
                                </div>
                            </Grid>
                        </Grid>
                    </form>
                </Container>
            </Paper>
        );
    }
}

export default withAuth(LoginForm);