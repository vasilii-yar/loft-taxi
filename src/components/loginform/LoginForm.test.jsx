import React from 'react';
import {render} from "@testing-library/react";
import LoginForm from "./LoginForm";
import {AuthProvider} from "../../util/AuthContext";
import '@testing-library/jest-dom';
import renderer from "react-test-renderer";

const mockNavigateTo = jest.fn();
jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
    Map: () => ({})
}));

describe("LoginForm", () => {
    it("renders correctly", () => {
        const {getByLabelText, getByText, getAllByText} = render(
            <AuthProvider>
                <LoginForm navigateTo={mockNavigateTo}/>
            </AuthProvider>
        );

        expect(getByLabelText("Пользователь")).toBeInTheDocument();
        expect(getByLabelText("Пароль")).toBeInTheDocument();
        expect(getByText("Зарегистрируйтесь")).toBeInTheDocument();
        expect(getAllByText("Войти").length.toString()).toMatch("2");
    })

    it("renders correctly check by snapshot", () => {
        const tree = renderer.create(
            <AuthProvider>
                <LoginForm navigateTo={mockNavigateTo}/>
            </AuthProvider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
});