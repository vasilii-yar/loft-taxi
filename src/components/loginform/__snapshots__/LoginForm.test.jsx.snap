// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly check by snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h4
      className="MuiTypography-root login-header MuiTypography-h4 MuiTypography-gutterBottom"
    >
      Войти
    </h4>
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Новый пользователь? 
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Зарегистрируйтесь
            </a>
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="login"
            >
              Пользователь
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="login"
                name="login"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="email"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="password"
            >
              Пароль
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="login-button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Войти
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
